From cd96caf9650dc6235d3ba700ab85f779e9709e8a Mon Sep 17 00:00:00 2001
From: Christoph Honal <christoph.honal@web.de>
Date: Mon, 24 Aug 2020 05:01:29 +0200
Subject: [PATCH] NRF5x: Enable deep sleep in BLE idle hook

---
 .../TARGET_NRF5x/NRFCordioHCIDriver.cpp                  | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/NRFCordioHCIDriver.cpp b/features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/NRFCordioHCIDriver.cpp
index ccccdf65886..3ee2357190b 100644
--- a/features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/NRFCordioHCIDriver.cpp
+++ b/features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/NRFCordioHCIDriver.cpp
@@ -200,11 +200,12 @@ static void idle_hook(void)
 	}
 
     // critical section to complete sleep with locked deepsleep
-    core_util_critical_section_enter();
-    sleep_manager_lock_deep_sleep();
+    // core_util_critical_section_enter();
+    // sleep_manager_lock_deep_sleep();
+#   warning Unofficial patch: Enable deep sleep in BLE idle hook - may be unstable
     sleep();
-    sleep_manager_unlock_deep_sleep();
-    core_util_critical_section_exit();
+    // sleep_manager_unlock_deep_sleep();
+    // core_util_critical_section_exit();
 }
 
 NRFCordioHCIDriver::NRFCordioHCIDriver(CordioHCITransportDriver& transport_driver) : cordio::CordioHCIDriver(transport_driver), _is_init(false), _stack_buffer(NULL)
